### ISSUE TITLE
IDISP001 does not warn for IDisposableAsync leaks

### ISSUE THREAD

**Issue by Kuinox** [DotNetAnalyzers/IDisposableAnalyzers#375]
```csharp
public class Program {
    public static async Task Main() {
        var a = new AsyncDiposable(); // nothing
        var b = new Diposable(); // warning IDISP001: Dispose created

        await Task.Delay(5000);
    }
    static AsyncDiposable LoadAsync() => new AsyncDiposable();
 }

public sealed class Diposable : IDisposable
{
    public void Dispose() {}
}

public sealed class AsyncDiposable : IAsyncDisposable
{
    public ValueTask DisposeAsync() => new ValueTask();
}
```
I would expect that AsyncDisposable has a warn there. 
Is IDisposableAsync not supported ?

**Comment by JohanLarsson** [comment]
The support for `IAsyncDisposable` is still terrible in this analyzer.
Thanks for reporting and very nice issue!
Sorry about the bug.

**Comment by Kuinox** [comment]
Good to know ^^
Don't worry I opened these issues to have a trace/something to do if I'm bored

**Comment by JohanLarsson** [comment]
Never hesitate to write issues, the more the better. Many small are best for me.

**Comment by erichiller** [comment]
To add to the above, `async` Methods returning disposables are not marked.

For example, [`NpgsqlConnection` is `IDisposable` and `IAsyncDisposable`](https://www.npgsql.org/doc/api/Npgsql.NpgsqlConnection.html#Npgsql_NpgsqlConnection_Dispose_System_Boolean_) and when returned from a synchronous method it is marked, but not from asynchronous methods.

For example, in the below `OpenConnectionAsync` should also be marked the same as the other lines.
![image](https://github.com/DotNetAnalyzers/IDisposableAnalyzers/assets/5379325/06c70f20-6ed3-4643-88e4-c8f3793b0236)
